gen/tests/test_validation.py

```diff
@@ -733,3 +733,14 @@ def test_validate_custom_checks():
             'node check names: node-check-1, node-check-2.'
         ),
     )
+
+
+def test_fault_domain_disabled():
+    arguments = make_arguments(new_arguments={
+        'fault_domain_detect_filename': 'gen/fault-domain-detect/aws.sh'
```

Comment:
This is a relative path, which means it depends on the test environment, which may make the test brittle. Let's use `pkg_resources.resource_filename()` to make sure we can always find it. Example use here: https://github.com/dcos/dcos/blob/0de96834fdebf56e8c8ae1ca77f4ef62b4885a5b/gen/tests/utils.py#L21